provider/pkg/provider/provider.go

```diff
@@ -1430,7 +1442,7 @@ func (k *kubeProvider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (
 		}
 
 		// Validate the object according to the OpenAPI schema for its GVK.
-		err = openapi.ValidateAgainstSchema(resources, newInputs)
+		err = openapi.ValidateAgainstSchema(k.getResourcesLazy(), resources, newInputs)
```

Comment:
Is this effectively the same as before? Would there be a perf consideration here?

Reply:
This still needs a rework so we don't unnecessarily fetch the OpenAPI spec again, since we already call `k.getResources()`. I've just done it like this for now just to generate a schema to work from. Thanks for the initial review and flagging!

Reply:
Our implementation of OpenAPIResourcesGetter now returns a list of resources that have already been pre-fetched earlier for additional validation outside of the upstream helper function.